core/loop.gms

```diff
@@ -149,6 +149,16 @@ o_negitr_disc_cons_drInt_reg(iteration,regi) =
     + sum(ttot$(ttot.val eq 2100), vm_cons.l(ttot,regi) * qm_budget.m(ttot,regi)/ (qm_budget.m("2005",regi) + 1.e-8) * ( pm_ttot_val(ttot)- pm_ttot_val(ttot-1) ) * 0.5 )
 ;
 
+***--------------------------------------------------------------------------
+***         Save GDX with solution before postsolve
+***--------------------------------------------------------------------------
+execute_unload 'fulldata_prepostsolve';
```

Comment:
prepostsolve is a bit like Ostwestfalen.

Reply:
And much like Ostwestfalen contains Bielefeld - an imaginary place people still claim to have visited - the current magicc6 integration contains only nebulous statements with regards to how it obtains it's remind emission data

While the `_prepostsolve` name can be improved upon, the more pressing issue - that i just now realize - is that we're doing a gdx dump *every iteration*, regardless of the `15_climate` module configuration. This needs to be limited, right? However I do not feel proficient enough in GAMS to do this myself. Please help @gabriel-abrahao @LaviniaBaumstark

Reply:
Sidenote: Thanks to @0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q who explained me that the emission data dump happens in [`./core/magicc.c`](https://github.com/remindmodel/remind/blob/develop/core/magicc.gms)

Reply:
MAGICC6 gets its data [from the file written here](https://github.com/remindmodel/remind/blob/develop/core/magicc.gms#L121-L191). Agree that you best put that behind some kind of if-clause like 
```
if(sameas("%climate%","magicc7_ar7"), p46_startInIteration = 0);
```
Of course, replace the p46 stuff by the code you want to execute. But there is also the
```
$Ifi "%carbonprice%" == "temperatureNotToExceed" $include "./modules/45_carbonprice/temperatureNotToExceed/realization.gms"
```
construction and the
```
$if %carbonprice% == "off"      pm_macSwitch(emiMacMagpie) = 0;
```
way and I never really understood what to use.

Reply:
> While the `_prepostsolve` name can be improved upon, the more pressing issue - that i just now realize - is that we're doing a gdx dump _every iteration_, regardless of the `15_climate` module configuration. This needs to be limited, right?

Just put it inside `./modules/15_climate/magicc7_ar6_whatever/postsolve.gms`.

Reply:
in generall, I agre wit Michaja. If we need the gdx dump in many realizations and wtiht the same information (at the same time during solution) we might do it once at a central place